### ISSUE TITLE
Question: To what extent does watermelon block the main thread?

### ISSUE THREAD

**Issue by mfbx9da4** [Nozbe/WatermelonDB#1378]
Say for example the below code takes 100ms

```typescript
await collection.query(Q.where('id', Q.oneOf(ids)).fetch()
```

Some of that will be blocking the main JS thread e.g. updating the records cache. Some of the time will be idle waiting for the resolved microtask to be enqueued. How can I find out how much of that 100ms is spent blocking?

Similarly, how can I measure the blocking time of `sync.applyRemoteChanges()`?

**Comment by radex** [comment]
> How can I find out how much of that 100ms is spent blocking?

use a profiler :)

**Comment by radex** [comment]
no, a real profiler. If you want to see threading issues (ie, this extends outside of JS), it has to be a native profiler, like Instruments.app

**Comment by mfbx9da4** [comment]
I have been trying hards to use instruments but it doesn't really help me in knowing if the JS thread is blocked because all the JS functions are just [memory addresses like this](https://stackoverflow.com/questions/58596686/how-to-profile-react-native-source-code-using-xcode-instruments-time-profiler)? 

I've had some success using Safari > Develop > {Simulator} > JSContext but it's kind of awkward to use. 

How do you do measure blocking time out of interest? 😄
